### ISSUE TITLE
No more than one IMAP Trigger Node working

### ISSUE THREAD

**Issue by KleSecGmbH** [n8n-io/n8n#7190]
If i create a Workflow with an IMAP Trigger Node, everything works fine. As soon as I create another Workflow using the Node, it won't Trigger.

**To Reproduce**
Create multiple IMAP Trigger in multiple Workflows

**Environment**

4 Tested Systems, all cloudron based. Newest N8N package. Same behavior.
IMAP Server: OVH Mail Basic

**Comment by Joffcom** [comment]
Hey @KleSecGmbH 

That is interesting I currently have 3 imap nodes working on my local n8n install.

Can you share a bit more? Are you using different email accounts or the same account and which n8n version are you running?

**Comment by KleSecGmbH** [comment]
Hey there, thanks for the quick response. This issue first appeared 2 days ago while we were working on a Project. This Project was running an n8n Workflow with an IMAP Trigger node for over a month without any problems whatsoever (Mailserver OVH). It was running version 1.3.1 back then. The original plan was to use the IMAP feature on cloudron, so i decided to create a new workflow with another imap node. Whilst listening for events on the IMAP node everything works as expected. But if only the workflow is active, it will never get triggered no matter how many emails i send. After checking this up online i decided to update N8N to 1.6.1, deleted the workflow and added a new one. Same thing, but all the time the mails in the mailbox are beeing marked as read. After cloudrons IMAP didn´t work, I decided to use another OVH Mailbox, but still no luck at all. After that i gave up and implemented the needed features by Webhook and REST but it would be awesome to get this working for future Projects. Please let me know what information i can provide to you in terms of logs on cloudron.

**Comment by Joffcom** [comment]
Hey @KleSecGmbH

So I guess the first part is was it multiple triggers for one mailbox or multiple triggers for different mailboxes?

Can you enable debug logging in n8n and let it run then send over the log so I can take a look? All of my flows are for different gmail and Google workplace accounts.

**Comment by KleSecGmbH** [comment]
I just tested it again wit debugging on. This is the full output i get when only the Workflow is active and i send an email to the IMAP account:

```
Sep 18 19:39:54 2023-09-18T17:39:54.854Z | debug | Wait tracker querying database for waiting executions "{ file: 'WaitTracker.js', function: 'getWaitingExecutions' }"

Sep 18 19:39:58 2023-09-18T17:39:58.876Z | debug | Querying for new messages on node "EmailReadImap" "{\n searchCriteria: [\n 'UNSEEN',\n [ 'UID', '58:*' ],\n [ 'UID', '58:*' ],\n [ 'UID', '58:*' ],\n [ 'UID', '58:*' ]\n ],\n file: 'EmailReadImapV2.node.js',\n function: 'onmail'\n}"

Sep 18 19:39:59 2023-09-18T17:39:59.068Z | verbose | Email Read Imap:update 27 "{\n uid: 27,\n modseq: '50',\n file: 'EmailReadImapV2.node.js',\n function: 'onupdate'\n}" 
```


If i click on Execute Workflow the following happens:

```
Sep 18 19:43:02 2023-09-18T17:43:02.106Z | verbose | Execution for workflow TEST was assigned id 178 "{\n executionId: '178',\n file: 'WorkflowRunner.js',\n function: 'runMainProcess'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.106Z | verbose | Execution for workflow TEST was assigned id 178 "{\n executionId: '178',\n file: 'WorkflowRunner.js',\n function: 'runMainProcess'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.109Z | debug | Execution ID 178 will run executing all nodes. "{\n executionId: '178',\n file: 'WorkflowRunner.js',\n function: 'runMainProcess'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.110Z | verbose | Workflow execution started "{\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js',\n function: 'processRunExecutionData'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.110Z | debug | Executing hook (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'workflowExecuteBefore'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.111Z | debug | Send data of type "executionStarted" to editor-UI "{\n dataType: 'executionStarted',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.112Z | debug | Start processing node "Email Trigger (IMAP)" "{\n node: 'Email Trigger (IMAP)',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.113Z | debug | Executing hook on node "Email Trigger (IMAP)" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteBefore'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.113Z | debug | Send data of type "nodeExecuteBefore" to editor-UI "{\n dataType: 'nodeExecuteBefore',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.113Z | debug | Running node "Email Trigger (IMAP)" started "{\n node: 'Email Trigger (IMAP)',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:02 2023-09-18T17:43:02.127Z | debug | Loaded static data for node "EmailReadImap" "{\n staticData: {},\n file: 'EmailReadImapV2.node.js',\n function: 'trigger'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.104Z | debug | Querying for new messages on node "EmailReadImap" "{\n searchCriteria: [\n 'UNSEEN',\n [ 'UID', '58:*' ],\n [ 'UID', '58:*' ],\n [ 'UID', '58:*' ],\n [ 'UID', '58:*' ],\n [ 'UID', '58:*' ]\n ],\n file: 'EmailReadImapV2.node.js',\n function: 'onmail'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.298Z | verbose | Email Read Imap:update 28 "{\n uid: 28,\n modseq: '52',\n file: 'EmailReadImapV2.node.js',\n function: 'onupdate'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.328Z | verbose | Email Read Imap:update 28 "{\n uid: 28,\n modseq: '52',\n flags: [ '\\\\Seen' ],\n file: 'EmailReadImapV2.node.js',\n function: 'onupdate'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.328Z | debug | Running node "Email Trigger (IMAP)" finished successfully "{\n node: 'Email Trigger (IMAP)',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.329Z | debug | Executing hook on node "Email Trigger (IMAP)" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteAfter'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.329Z | debug | Send data of type "nodeExecuteAfter" to editor-UI "{\n dataType: 'nodeExecuteAfter',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.345Z | debug | Start processing node "Create Year Folder1" "{\n node: 'Create Year Folder1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.345Z | debug | Executing hook on node "Create Year Folder1" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteBefore'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.346Z | debug | Send data of type "nodeExecuteBefore" to editor-UI "{\n dataType: 'nodeExecuteBefore',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.346Z | debug | Running node "Create Year Folder1" started "{\n node: 'Create Year Folder1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:19 2023-09-18T17:43:19.370Z | debug | Proxying request to axios "{ file: 'NodeExecuteFunctions.js', function: 'proxyRequestToAxios' }"
Sep 18 19:43:19 2023-09-18T17:43:19.405Z | debug | Email Read Imap: Shutting down workflow - connected closed "{ file: 'EmailReadImapV2.node.js' }"
Sep 18 19:43:20 2023-09-18T17:43:20.226Z | debug | Request proxied to Axios failed "{\n status: 405,\n file: 'NodeExecuteFunctions.js',\n function: 'proxyRequestToAxios'\n}"
Sep 18 19:43:20 2023-09-18T17:43:20.226Z | debug | Running node "Create Year Folder1" finished successfully "{\n node: 'Create Year Folder1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:20 2023-09-18T17:43:20.227Z | debug | Executing hook on node "Create Year Folder1" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteAfter'\n}"
Sep 18 19:43:20 2023-09-18T17:43:20.227Z | debug | Send data of type "nodeExecuteAfter" to editor-UI "{\n dataType: 'nodeExecuteAfter',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:20 2023-09-18T17:43:20.228Z | debug | Start processing node "Create Month Folder1" "{\n node: 'Create Month Folder1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:20 2023-09-18T17:43:20.228Z | debug | Executing hook on node "Create Month Folder1" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteBefore'\n}"
Sep 18 19:43:20 2023-09-18T17:43:20.228Z | debug | Send data of type "nodeExecuteBefore" to editor-UI "{\n dataType: 'nodeExecuteBefore',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:20 2023-09-18T17:43:20.229Z | debug | Running node "Create Month Folder1" started "{\n node: 'Create Month Folder1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:20 2023-09-18T17:43:20.243Z | debug | Proxying request to axios "{ file: 'NodeExecuteFunctions.js', function: 'proxyRequestToAxios' }"
Sep 18 19:43:21 2023-09-18T17:43:21.144Z | debug | Request proxied to Axios failed "{\n status: 405,\n file: 'NodeExecuteFunctions.js',\n function: 'proxyRequestToAxios'\n}"
Sep 18 19:43:21 2023-09-18T17:43:21.145Z | debug | Running node "Create Month Folder1" finished successfully "{\n node: 'Create Month Folder1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:21 2023-09-18T17:43:21.146Z | debug | Executing hook on node "Create Month Folder1" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteAfter'\n}"
Sep 18 19:43:21 2023-09-18T17:43:21.146Z | debug | Send data of type "nodeExecuteAfter" to editor-UI "{\n dataType: 'nodeExecuteAfter',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:21 2023-09-18T17:43:21.146Z | debug | Start processing node "Wait for Folder Create1" "{\n node: 'Wait for Folder Create1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:21 2023-09-18T17:43:21.147Z | debug | Executing hook on node "Wait for Folder Create1" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteBefore'\n}"
Sep 18 19:43:21 2023-09-18T17:43:21.147Z | debug | Send data of type "nodeExecuteBefore" to editor-UI "{\n dataType: 'nodeExecuteBefore',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:21 2023-09-18T17:43:21.147Z | debug | Running node "Wait for Folder Create1" started "{\n node: 'Wait for Folder Create1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:26 2023-09-18T17:43:26.150Z | debug | Running node "Wait for Folder Create1" finished successfully "{\n node: 'Wait for Folder Create1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:26 2023-09-18T17:43:26.151Z | debug | Executing hook on node "Wait for Folder Create1" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteAfter'\n}"
Sep 18 19:43:26 2023-09-18T17:43:26.151Z | debug | Send data of type "nodeExecuteAfter" to editor-UI "{\n dataType: 'nodeExecuteAfter',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:26 2023-09-18T17:43:26.166Z | debug | Start processing node "Upload File1" "{\n node: 'Upload File1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:26 2023-09-18T17:43:26.166Z | debug | Executing hook on node "Upload File1" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteBefore'\n}"
Sep 18 19:43:26 2023-09-18T17:43:26.167Z | debug | Send data of type "nodeExecuteBefore" to editor-UI "{\n dataType: 'nodeExecuteBefore',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:26 2023-09-18T17:43:26.167Z | debug | Running node "Upload File1" started "{\n node: 'Upload File1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:26 2023-09-18T17:43:26.189Z | debug | Proxying request to axios "{ file: 'NodeExecuteFunctions.js', function: 'proxyRequestToAxios' }"
Sep 18 19:43:27 2023-09-18T17:43:27.709Z | debug | Running node "Upload File1" finished successfully "{\n node: 'Upload File1',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js'\n}"
Sep 18 19:43:27 2023-09-18T17:43:27.710Z | debug | Executing hook on node "Upload File1" (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'nodeExecuteAfter'\n}"
Sep 18 19:43:27 2023-09-18T17:43:27.710Z | debug | Send data of type "nodeExecuteAfter" to editor-UI "{\n dataType: 'nodeExecuteAfter',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"
Sep 18 19:43:27 2023-09-18T17:43:27.711Z | verbose | Workflow execution finished successfully "{\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecute.js',\n function: 'processSuccessExecution'\n}"
Sep 18 19:43:27 2023-09-18T17:43:27.711Z | debug | Executing hook (hookFunctionsSave) "{\n executionId: '178',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'workflowExecuteAfter'\n}"
Sep 18 19:43:27 2023-09-18T17:43:27.711Z | debug | Save execution data to database for execution ID 178 "{\n executionId: '178',\n workflowId: 'quv2TkjfHdsBp4Zc',\n finished: true,\n stoppedAt: 2023-09-18T17:43:27.711Z,\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'workflowExecuteAfter'\n}"
Sep 18 19:43:27 2023-09-18T17:43:27.854Z | debug | Executing hook (hookFunctionsPush) "{\n executionId: '178',\n sessionId: 'yx0z16lbdwh',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'workflowExecuteAfter'\n}"
Sep 18 19:43:27 2023-09-18T17:43:27.855Z | debug | Save execution progress to database for execution ID 178 "{\n executionId: '178',\n workflowId: 'quv2TkjfHdsBp4Zc',\n file: 'WorkflowExecuteAdditionalData.js',\n function: 'workflowExecuteAfter'\n}"
Sep 18 19:43:27 2023-09-18T17:43:27.855Z | debug | Send data of type "executionFinished" to editor-UI "{\n dataType: 'executionFinished',\n sessionId: 'yx0z16lbdwh',\n file: 'abstract.push.js',\n function: 'send'\n}"`
```
which is the expected behavior. Why doesn´t the first one not trigger the workflow? It seems to have recognized the mail, doesn´t it?

**Comment by Joffcom** [comment]
Hey @KleSecGmbH,

Generally when I am testing I don't use execute workflow as it doesn't fetch the same emails so leaving it enabled to do its thing is the best bet.

I would have expected more output, do you have a screenshot of the workflow you are running? We have seen a few issues with the imap node but reproducing the issues has been tricky.

**Comment by Joffcom** [comment]
Closing for now as not able to reproduce, if this is still an issue we can re-open it.

**Comment by webkp** [comment]
I encountered what I think is the same problem today. 

Everything worked for months. Suddenly it didn't seem to. 

During tests I also noticed that messages were marked as read but did not trigger the workflow any further. 

On closer inspection, I noticed in the debug logs that n8n adds a UID range to the search criteria. 

In my case, there were changes to the email infrastructure a few days ago and the smallest UID stored by n8n was much larger than that of the newly arrived messages. 

I think the same can be seen in @KleSecGmbH logs from above:
```
Sep 18 19:43:19 2023-09-18T17:43:19.104Z | debug | Querying for new messages on node "EmailReadImap" "{ searchCriteria: [ 'UNSEEN', [ 'UID', '58:*' ], [ 'UID', '58:*' ], [ 'UID', '58:*' ], [ 'UID', '58:*' ], [ 'UID', '58:*' ] ], file: 'EmailReadImapV2.node.js', function: 'onmail'}"
Sep 18 19:43:19 2023-09-18T17:43:19.298Z | verbose | Email Read Imap:update 28 "{ uid: 28, modseq: '52', file: 'EmailReadImapV2.node.js', function: 'onupdate'}"
```
Search criteria has an added "[ 'UID', '**58**:*' ]" clause while the UID from the message is 28 ("{ uid: **28**, modseq: '52', fla...").

BTW: I wonder why the UID filter gets added so many times. It was the same in my logs @Joffcom 
While I think it's not a problem since the underlying node-imap module takes multiple criterias as AND if OR is not explicitly used it still seems a bit messy.

**Solution**
Unfortunately, duplicating the workflow did not help, as staticData is presumably also copied? (not sure as I'm not a Node dev)

Only creating a new workflow without copying the IMAP node from the old one but adding the IMAP node from scratch worked for me (copying the other non IMAP-nodes was possible).

I think I might ran into the problem mentioned in a comment of the IMAP node:

https://github.com/n8n-io/n8n/blob/dd405700568acdc893358ef239e8f8637da66307/packages/nodes-base/nodes/EmailReadImap/v2/EmailReadImapV2.node.ts#L566-L577

It might be helpful if UIDValidity could be checked @Joffcom or give possibility to alter stored UID in the settings of the node to be able to reset without having to recreate the whole workflow.
